src/student/commands/list-student-schedules.command.ts

```diff
@@ -18,21 +18,26 @@ export class ListStudentSchedulesCommand {
       },
     };
 
-    const monitor = await this.prisma.monitor.findFirst({
+    const monitorings = await this.prisma.monitor.findMany({
       where: {
         student_id: user_id,
-        id_status: MonitorStatus.AVAILABLE,
+        id_status: { in: [MonitorStatus.AVAILABLE, MonitorStatus.DONE] },
       },
       include: {
         student: studentInclude,
       },
     });
 
-    let orStatement = [{ student_id: user_id }, { monitor_id: monitor?.id }];
+    const orStatement: ({ student_id: number } | { monitor_id: number })[] = [
+      { student_id: user_id },
+    ];
+    monitorings.forEach((monitoring) =>
+      orStatement.push({ monitor_id: monitoring?.id }),
+    );
```

Comment:
Usando o operador IN aqui ia ficar mais simples:
```suggestion
    const orStatement = [
      { student_id: user_id },
      { monitor_id: { in: monitorings.map(monitor => monitor.id) } }
    ];
```

Reply:
Aí no lugar do shift e do splice você pode acessar os índices 0 e 1 de acordo com o caso, assim fica mais performático.